packages/common/http/src/transfer_cache.ts

```diff
@@ -100,10 +103,13 @@ export function transferCacheInterceptorFn(
   // In the following situations we do not want to cache the request
   if (
     !isCacheActive ||
+    requestOptions === false ||
     // POST requests are allowed either globally or at request level
     (requestMethod === 'POST' && !globalOptions.includePostRequests && !requestOptions) ||
     (requestMethod !== 'POST' && !ALLOWED_METHODS.includes(requestMethod)) ||
-    requestOptions === false || //
+    // Do not cache request that require authorization when includeRequestsWithAuthHeaders is falsey
+    (!globalOptions.includeRequestsWithAuthHeaders &&
+      (req.headers.has('authorization') || req.headers.has('proxy-authorization'))) ||
```

Comment:
nit: I'd propose moving the logic that checks for specific headers into a separate function:
```suggestion
    (!globalOptions.includeRequestsWithAuthHeaders && hasAuthHeaders(req)) ||
```